src/test/scala/CoreServerTests.scala

```diff
@@ -74,13 +73,12 @@ class CoreServerTest extends WordSpec with Matchers with ScalatestRouteTest {
 
       var handler: VerificationJobHandler = null
       "be able to execute 'verify()' without exceptions" in {
-        handler = core.verify(file, noCacheBackend, ast)
+        handler = core.verify(verificationError_file, noCache_backend, verificationError_ast)
         assert(handler != null)
       }
 
       "be able to have 'verify()' return a JobHandler with non-negative id when executed" in {
```

Comment:
I would rephrase the messages to say, e.g., "check that [if A then] B" rather than "be able to have [B] if A".

Reply:
Throughout the test suit the idea if these description was to have the test subject (usually, a method or some result) at the beginning of the phrase, and then what the expected bahaviour is. I'll generally leave it this way.

In this particular case "when executed" indeed shouldn't be there.